otherlibs/dune-rpc/dune_rpc.mli

```diff
@@ -82,25 +82,56 @@ module V1 : sig
       }
   end
 
-  module Error : sig
+  module Diagnostic : sig
+    type severity =
+      | Error
+      | Warning
+
+    module Promotion : sig
+      type t =
+        { in_build : string
+        ; in_source : string
+        }
+    end
+
     type t =
       { target : string option
       ; message : string
       ; loc : Loc.t option
+      ; severity : severity option
+      ; promotion : Promotion.t option
```

Comment:
Why are promotions in diagnostic now? It seems unrelated, at least from a user's point of view.

Reply:
I think we came up with this idea in a meeting with @jeremiedimino. The idea is that a promotion should be counted as an error because it suggests that some `diff` action failed the build.